server/sonar-server/src/main/java/org/sonar/server/webhook/WebHooks.java

```diff
@@ -44,35 +46,44 @@
   final class Analysis {
     private final String projectUuid;
     private final String ceTaskUuid;
+    private final String analysisUuid;
 
-    public Analysis(String projectUuid, String ceTaskUuid) {
+    public Analysis(String projectUuid, String analysisUuid, @Nullable  String ceTaskUuid) {
       this.projectUuid = requireNonNull(projectUuid, "projectUuid can't be null");
-      this.ceTaskUuid = requireNonNull(ceTaskUuid, "ceTaskUuid can't be null");
+      this.analysisUuid = requireNonNull(analysisUuid, "analysisUuid can't be null");
+      this.ceTaskUuid = ceTaskUuid;
     }
 
     public String getProjectUuid() {
       return projectUuid;
     }
 
+    @CheckForNull
     public String getCeTaskUuid() {
       return ceTaskUuid;
     }
 
+    public String getAnalysisUuid() {
+      return analysisUuid;
+    }
+
     @Override
     public boolean equals(Object o) {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof Analysis)) {
         return false;
       }
       Analysis analysis = (Analysis) o;
-      return projectUuid.equals(analysis.projectUuid) && ceTaskUuid.equals(analysis.ceTaskUuid);
+      return Objects.equal(projectUuid, analysis.projectUuid) &&
+        Objects.equal(ceTaskUuid, analysis.ceTaskUuid) &&
+        Objects.equal(analysisUuid, analysis.analysisUuid);
     }
 
     @Override
     public int hashCode() {
-      return Objects.hash(projectUuid, ceTaskUuid);
+      return Objects.hashCode(projectUuid, ceTaskUuid, analysisUuid);
```

Comment:
why depend on Guava's `Object` class when JDK provides the replacement for it?
(and even more, why change it when it's already in place?)

Reply:
Just because I used the generator of equals and hashcode.
Reverting.